iidm/iidm-api/src/main/java/com/powsybl/iidm/network/ValidationUtil.java

```diff
@@ -383,13 +518,302 @@ public static void checkPhaseTapChangerRegulation(Validable validable, PhaseTapC
+     * Use {@link #checkSvcRegulation(Validable, Terminal, double, double, RegulationMode, Network)}
+     */
+    @Deprecated(since = "4.4.0")
+    public static void checkSvcRegulator(Validable validable, double voltageSetpoint, double reactivePowerSetpoint, StaticVarCompensator.RegulationMode regulationMode) {
+        if (regulationMode == null) {
+            throw new ValidationException(validable, "Regulation mode is invalid");
+        }
+        switch (regulationMode) {
+            case VOLTAGE:
+                if (Double.isNaN(voltageSetpoint)) {
+                    throw createInvalidValueException(validable, voltageSetpoint, VOLTAGE_SETPOINT);
+                }
+                break;
+
+            case REACTIVE_POWER:
+                if (Double.isNaN(reactivePowerSetpoint)) {
+                    throw createInvalidValueException(validable, reactivePowerSetpoint, REACTIVE_POWER_SETPOINT);
+                }
+                break;
+
+            case OFF:
+                // nothing to check
+                break;
+
+            default:
+                throw new AssertionError();
+        }
+
+    }
+
+    /**
+     * For validating static Var compensator control.
+     * Regulating Terminal is forced to be validated.
+     */
+    public static void checkSvcRegulation(Validable validable, Terminal regulatingTerminal, double voltageSetpoint,
+        double reactivePowerSetpoint, RegulationMode regulationMode, Network network) {
+        checkSvcRegulation(validable, regulatingTerminal, voltageSetpoint, reactivePowerSetpoint, regulationMode, network, true);
+    }
+
+    /**
+     * For validating static Var compensator control.
+     * User can select if regulating terminal should be validated.
+     * Depending on RegulationMode (OFF, VOLTAGE, REACTIVE_POWER) static Var compensator will control voltage or reactive power.
+     * When RegulationMode is OFF there is no validation.
+     * When RegulationMode is VOLTAGE, voltage setpoint and regulating terminal must be defined and valid.
+     * When RegulationMode is REACTIVE_POWER reactive power setpoint and regulating terminal must be defined and valid.
+     */
+    public static void checkSvcRegulation(Validable validable, Terminal regulatingTerminal, double voltageSetpoint,
+        double reactivePowerSetpoint, RegulationMode regulationMode, Network network, boolean validateRegulatingTerminal) {
+        if (regulationMode == null) {
+            throw new ValidationException(validable, "Regulation mode is invalid");
+        }
+        if (regulationMode == StaticVarCompensator.RegulationMode.VOLTAGE) {
+            ValidationUtil.checkContinuousVoltageControl(validable, regulatingTerminal, voltageSetpoint, true, network, validateRegulatingTerminal);
+        } else if (regulationMode == StaticVarCompensator.RegulationMode.REACTIVE_POWER) {
+            ValidationUtil.checkReactivePowerControl(validable, regulatingTerminal, reactivePowerSetpoint, true, network);
```

Comment:
You don't have to specify `ValidationUtil` in front of methods of same class

Reply:
Done